app/services/samples/create_service.rb

```diff
@@ -14,7 +14,17 @@ def initialize(user = nil, project = nil, params = {})
     def execute
       authorize! @project, to: :create_sample? unless @project.nil?
 
-      sample.save
+      if sample.save
+        @project.namespace.create_activity key: 'namespaces_project_namespace.samples.create',
+                                           owner: current_user,
+                                           parameters:
+                                            {
+                                              sample_id: sample.id,
+                                              sample_name: sample.puid,
```

Comment:
**nitpick (blocking):** Can this be renamed to `sample_puid`

Reply:
Updated in 032cb1c08ee993dee5d18b74ef198b03bafe07b3